Crucible.GameAPI/CrucibleQuest.cs

```diff
@@ -0,0 +1,208 @@
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+// Crucible is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+// You should have received a copy of the GNU Lesser General Public License
+// along with Crucible; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+// </copyright>
+
+namespace RoboPhredDev.PotionCraft.Crucible.GameAPI
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using global::PotionCraft.Core.ValueContainers;
+    using global::PotionCraft.QuestSystem;
+    using HarmonyLib;
+    using UnityEngine;
+
+    /// <summary>
+    /// Provides a stable API for working with PotionCraft <see cref="Quest"/>s.
+    /// </summary>
+    public class CrucibleQuest
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CrucibleQuest"/> class using the provided <see cref="Quest"/>.
+        /// </summary>
+        /// <param name="quest">The <see cref="Quest"/> to use as a base.</param>
+        public CrucibleQuest(Quest quest)
+        {
+            this.Quest = quest;
+        }
+
+        /// <summary>
+        /// Gets or sets the base <see cref="Quest"/>.
+        /// </summary>
+        public Quest Quest { get; set; }
+
+        /// <summary>
+        /// Gets a value indicating whether the base quest is null.
+        /// </summary>
+        public bool IsNull => this.Quest == null;
+
+        /// <summary>
+        /// Gets or sets the uniqueId for this quest.
+        /// </summary>
+        public string ID
+        {
+            get => this.Quest.name;
+            set => this.Quest.name = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the karma reward for completing this quest.
+        /// </summary>
+        public int KarmaReward
+        {
+            get => this.Quest.karmaReward;
+            set => this.Quest.karmaReward = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the desired effects for the quest.
+        /// </summary>
+        public List<string> DesiredEffects
+        {
+            get => this.Quest.desiredEffects.Select(e => e.name).ToList();
+            set => this.Quest.desiredEffects = value.Select(e => CruciblePotionEffect.GetPotionEffectByID(e))
+                                                    .Select(e => e.PotionEffect)
+                                                    .ToArray();
+        }
+
+        /// <summary>
+        /// Gets or sets the minimum and maximum chapter needed to encounter this quest.
+        /// </summary>
+        public (int, int) MinMaxChapters
```

Comment:
Lets avoid tupple return types and just make 2 properties for MinChapter and MaxChapter.  Tupple work is a bit advanced for the skill set that this API is aimed at, and I am worried people wont know how to set just one.

Reply:
Alternatively, keep the tupple property, but add MinChapter and MaxChapter properties alongside it.

Reply:
Added two properties for min/max.